src/lib/mina_base/snapp_generators.ml

```diff
@@ -507,8 +507,22 @@ let gen_party_predicated_signed ?account_id ~ledger :
       ~increment_nonce:(Option.is_some account_id)
       ?account_id ~ledger ()
   in
-  let%map predicate = Account.Nonce.gen in
-  Party.Predicated.Poly.{ body; predicate }
+  (* use nonce from account in ledger *)
+  let pk = body.pk in
+  let account_id = Account_id.create pk body.token_id in
+  let account =
+    match Ledger.location_of_account ledger account_id with
+    | None ->
+        failwith "gen_party_predicated_signed: expected account to be in ledger"
```

Comment:
For new accounts and nonce=0, the transaction will be applied successfully. So maybe this shouldn't fail?
Also, I don't see this function's caller being used anywhere

Reply:
This was dead code -- `gen_party_predicated_fee_payer` replaced it.

The lookup failures should never occur, they're there in case there's a bug in the code.